tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/ILogger/DirectSubmission/LoggerFactoryIntegrationCommon.cs

```diff
@@ -0,0 +1,81 @@
+﻿// <copyright file="LoggerFactoryIntegrationCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Reflection;
+using System.Reflection.Emit;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Logging.DirectSubmission;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.ILogger.DirectSubmission
+{
+    internal static class LoggerFactoryIntegrationCommon<TLoggerFactory>
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(LoggerFactoryIntegrationCommon<TLoggerFactory>));
+
+        // ReSharper disable once StaticMemberInGenericType
+        // Internal for testing
+        internal static readonly Type ProviderInterfaces;
+
+        static LoggerFactoryIntegrationCommon()
+        {
+            // The ILoggerProvider type is in a different assembly to the LoggerFactory, so go via the ILogger type
+            // returned by CreateLogger
+            var loggerFactoryType = typeof(TLoggerFactory);
+            var abstractionsAssembly = loggerFactoryType.GetMethod("CreateLogger")?.ReturnType.Assembly;
```

Comment:
Is it possible we don't find this assembly? Should this static constructor have a try catch and disable the direct submission?

Reply:
> Is it possible we don't find this assembly?

I don't think that's possible? 🤔  

We're only in here because _Microsoft.Extensions.Logging_ has been loaded; we're in the `LoggerFactory` constructor. That constructor takes an `ILoggerProvider` (among other things) which is defined in the _Microsoft.Extensions.Logging.Abstractions_ library, so that must also be loaded.

That said it makes no sense to have `?` in this chain, given we access the assembly regardless in the next line.

If you still think there's a way that we could fail to find the assembly I don't mind adding a try catch for safety.

Reply:
I generally like the idea of no exceptions bubbling up from static constructors if you don't mind adding a safety net.

Reply:
Yeah, makes sense to be sure. Done in https://github.com/DataDog/dd-trace-dotnet/pull/1945/commits/878a8ebbca7a98f4c5481345b06944ac1bb9f71b